vernac/classes.ml

```diff
@@ -155,26 +152,28 @@ let new_instance ?(abstract=false) ?(global=false) ?(refine= !refine_instance)
     if generalize then CAst.make @@ CGeneralization (Implicit, Some AbsPi, tclass) 
     else tclass 
   in
-  let k, u, cty, ctx', ctx, len, imps, subst =
-    let impls, ((env', ctx), imps) = interp_context_evars env evars ctx in
-    let ctx = List.map (fun d -> Termops.map_rel_decl EConstr.Unsafe.to_constr d) ctx in
-    let c', imps' = interp_type_evars_impls ~impls env' evars tclass in
-    let c' = EConstr.Unsafe.to_constr c' in
+  let sigma, k, u, cty, ctx', ctx, len, imps, subst =
+    let _evd = ref sigma in
+    let impls, ((env', ctx), imps) = interp_context_evars env _evd ctx in
+    let c', imps' = interp_type_evars_impls ~impls env' _evd tclass in
+    let sigma = !_evd in
```

Comment:
I don't like underscore variables.
How about
```ocaml
let sigma = ref sigma in
let impls, ... = ... in
let c', ... = ... in
let sigma = !sigma in 
```

Reply:
That is temporal anyways, I have a patch migrating that not to use a ref. I like the underscore as it bugs people to fix it.